sympy/matrices/matrices.py

```diff
@@ -1581,6 +1583,12 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
+        num = sympify(num)
+        if not num.is_integer and (num >= 0) == True and not num % 1 == 0:
```

Comment:
Why cannot `num >= 0` be tested in the same place as `num < 0`? How does `not num % 1 == 0` exclude integer valued Floats?